bayesian_cnn_prometheus/learning/bayesian_detector.py

```diff
@@ -0,0 +1,127 @@
+from typing import Tuple, Dict
+
+import math
+from tensorflow.keras.callbacks import LearningRateScheduler, ModelCheckpoint
+from tensorflow_core.python.keras.optimizer_v2.adam import Adam
+
+from network.utils import *
+from .model.bayesian_vnet import bayesian_vnet
+
+
+class BayesianDetector:
+
+    def __init__(self, config: Dict):
+        preprocessing_config = config.get('../preprocessing')
+        self.config = config
+        self._input_shape: Tuple[int] = None
+        self._train_len: int = None
+        self._model = None
+
+        self._train_data = None
+        self._valid_data = None
+        self._test_data = None
+
+        # For creating a model
+        self._kernel_size = config.get('kernel_size')
+        self._activation = config.get('activation')
+        self._padding = config.get('padding')
+        self._prior_std = config.get('prior_std')
+
+        self._initial_epoch = config.get('initial_epoch')
+        self._kl_start_epoch = config.get('kl_start_epoch')
+        self._kl_alpha_increase_per_epoch = config.get('kl_alpha_increase_per_epoch')
+        self._kl_alpha = self._adjust_kl_alpha(config.get('kl_alpha'))
+        self._scale_factor = config.get('scale_factor')
+
+        self._weights_path = config.get('weights_path')
+        self._weights_dir = config.get('weights_dir')
+        self._checkpointer: ModelCheckpoint = None
+        self._scheduler: LearningRateScheduler = None
+        self._annealer: Callback = None
+        self._kl_start_epoch: int = config.get('kl_start_epoch')
+        self._kl_alpha_increase_per_epoch: float = config.get('kl_alpha_increase_per_epoch')
+
+        # For fitting
+        self._lr_decay_start_epoch = config.get('lr_decay_start_epoch')
+        self._epochs = config.get('epochs')
+        self._initial_epoch = config.get('initial_epoch')
+        self._valid_ds = config.get('valid_ds')
+
+    def _adjust_kl_alpha(self, kl_alpha: int):
+        if self._initial_epoch >= self._kl_start_epoch:
```

Comment:
Tutaj nie obsługujemy `None`, a może on spokojnie być, ponieważ z tego co widzę używasz do wyciągnięcia `get`.
Więc jak ktoś nie poda tej wartości w konfigu, to dostaniemy wyjątek 
![image](https://user-images.githubusercontent.com/21079319/141594277-f2379278-04d5-4c9b-b83a-aa6bc2b2bcb3.png)

Reply:
Okej, od razu wartości domyślne mogłyby być przechowywane w jednym mejscu, aby nie były zwykłymi liczbami w kodzie. zadziałam :)